src/main/js/components/TableView.jsx

```diff
@@ -76,12 +85,100 @@ class TableView extends React.Component {
 
     const {currentTableId} = this.state;
     const cols = this.tables.get(currentTableId).columns.models;
-    const view = this.calcVisibilityArray();
+    const view = cols.reduce((a, b) => f.merge({[b.id]: b.visible}, a), {});
     const savedViews = either(localStorage)
       .map(f.prop(["tableViews"]))
       .map(JSON.parse)
       .getOrElse({});
-    localStorage["tableViews"] = JSON.stringify(f.set([currentTableId,name], view, savedViews))
+    localStorage["tableViews"] = JSON.stringify(f.set([currentTableId, name], view, savedViews));
+  };
+
+  cellJumpError = msg => {
+    ActionCreator.showToast(
+      <div id="cell-jump-toast">{msg}</div>,
+      7000
+    )
+  };
+
+  checkGotoCellRequest = (loaded) => {
+    if (!this.pendingCellGoto) {
+      return;
+    }
+
+    const column = this.pendingCellGoto.column;
+    const columns = this.getCurrentTable().columns.models;
+    if (!this.checkIfColExists(columns, column)) {
+      return;
+    }
+    const {page} = this.pendingCellGoto;
+    if (loaded >= page || this.tableFullyLoaded) {
+      this.gotoCell(this.pendingCellGoto, loaded);
+    }
+  };
+
+  checkIfColExists = (columns, colId) => {
+    if (f.findIndex(f.matchesProperty("id", colId), columns) < 0) {
+      this.cellJumpError(i18n.t("table:jump.no_such_column", {col: colId}));
+      this.pendingCellGoto = null;
+      false;
+    } else {
+      return true;
+    }
+  };
+
+  estimateCellPage = row => 1 + Math.ceil((row - INITIAL_PAGE_SIZE) / PAGE_SIZE);
+
+  gotoCell = ({row, column, page, filter, ignore = false}, nPagesLoaded = 0) => {
```

Comment:
`row` is the row's ID isn't it? If so, could you please name it `rowId`. Some for `column`.